sw/device/lib/crypto/drivers/hmac.c

```diff
@@ -310,21 +313,25 @@ static void msg_fifo_write(const uint8_t *message, size_t message_len) {
   // TODO(#23191): Should we handle backpressure here?
   // Begin by writing a one byte at a time until the data is aligned.
   size_t i = 0;
-  for (; misalignment32_of((uintptr_t)(&message[i])) > 0 && i < message_len;
+  for (; misalignment32_of((uintptr_t)(&message[i])) > 0 &&
+         launder32(i) < message_len;
```

Comment:
this and the next loop dont to the HARDENED_CHECK after the loop; I suppose there is a reason I miss :)

Reply:
sry missed that i is not re-initialized. yes sg!